### DISCUSSION TITLE
Creating Final cut pro X library using cli.

### DISCUSSION THREAD

**Discussion by shell1986** [CommandPost/FCPCafe#351] [Final Cut Pro Questions]
Hello, I recently started thinking about optimizing the creation of libraries for specific projects.
They send me links to materials stored in cloud storages such as Yandex disk.
I created a bot that uploads them to my home Linux server.
I'm now thinking about the next level of optimization.
I want to create a script that would create proxy files using ffmpeg, and also create a library where all these files would be integrated.
I know how to encode video, but couldn't find any information on how to create a library outside of the Final Сut.
I would be glad for any help on this issue, maybe you know some libraries or someone has researched the file structure.
I found CurrentVersion.fcpevent and CurrentVersion.flexolibrary.
There is sqlite3 inside which does not have any information about the library and files.

**Comment by latenitefilms** [comment]
The easiest option would be to just generate a FCPXML and let FCPX create the library.
